pandas/core/groupby/groupby.py

```diff
@@ -641,12 +641,15 @@ def curried(x):
             # if we don't have this method to indicated to aggregate to
             # mark this column as an error
             try:
-                return self._aggregate_item_by_item(name, *args, **kwargs)
+                result = self._aggregate_item_by_item(name, *args, **kwargs)
```

Comment:
Can we just replace this function altogether with `_python_agg_general`? Conceptually it would seem like they do the same thing, though the former would actually be generic across `Series` and `DataFrame`

Reply:
I haven't checked specifically, but I'm hoping we can trim these before long